hal/armv7a/imx6ull/hal.c

```diff
@@ -127,12 +134,14 @@ static void hal_getMinOverlappedRange(addr_t start, addr_t end, mapent_t *entry,
 
 int hal_memoryGetNextEntry(addr_t start, addr_t end, mapent_t *entry)
 {
-	mapent_t tempEntry = { 0 };
+	mapent_t tempEntry;
 	mapent_t minEntry = { .start = (addr_t)-1, .end = 0 };
 
 	if (start == end)
 		return -1;
 
+	hal_memset(&tempEntry, 0, sizeof(tempEntry));
```

Comment:
It would be safer use `hal_memset()` 2 times:
```
	mapent_t tempEntry;
	mapent_t minEntry;

	if (start == end)
		return -1;

	hal_memset(&tempEntry, 0, sizeof(tempEntry));
	hal_memset(&minEntry, 0, sizeof(minEntry));
	minEntry.start = (addr_t)-1;
```

Reply:
Right, memcpy might be inserted, done